exchanges/coinbene/coinbene.go

```diff
@@ -0,0 +1,342 @@
+		common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout))
+	c.APIUrlDefault = coinbeneAPIURL
+	c.APIUrl = c.APIUrlDefault
+	c.Websocket = wshandler.New()
+	c.WebsocketURL = coinbeneWsURL
+	c.Websocket.Functionality = wshandler.WebsocketTickerSupported |
+		wshandler.WebsocketTradeDataSupported |
+		wshandler.WebsocketOrderbookSupported |
+		wshandler.WebsocketSubscribeSupported |
+		wshandler.WebsocketUnsubscribeSupported |
+		wshandler.WebsocketAuthenticatedEndpointsSupported
+	c.WebsocketResponseMaxLimit = exchange.DefaultWebsocketResponseMaxLimit
+	c.WebsocketResponseCheckTimeout = exchange.DefaultWebsocketResponseCheckTimeout
+	c.WebsocketOrderbookBufferLimit = exchange.DefaultWebsocketOrderbookBufferLimit
+}
+
+// Setup takes in the supplied exchange configuration details and sets params
+func (c *Coinbene) Setup(exch *config.ExchangeConfig) {
+	if !exch.Enabled {
+		c.SetEnabled(false)
+	} else {
+		c.Enabled = true
+		c.AuthenticatedAPISupport = exch.AuthenticatedAPISupport
+		c.AuthenticatedWebsocketAPISupport = exch.AuthenticatedWebsocketAPISupport
+		c.SetAPIKeys(exch.APIKey, exch.APISecret, "", false)
+		c.SetHTTPClientTimeout(exch.HTTPTimeout)
+		c.SetHTTPClientUserAgent(exch.HTTPUserAgent)
+		c.RESTPollingDelay = exch.RESTPollingDelay
+		c.Verbose = exch.Verbose
+		c.Websocket.SetWsStatusAndConnection(exch.Websocket)
+		c.BaseCurrencies = exch.BaseCurrencies
+		c.AvailablePairs = exch.AvailablePairs
+		c.EnabledPairs = exch.EnabledPairs
+		err := c.SetCurrencyPairFormat()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAssetTypes()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAutoPairDefaults()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetAPIURL(exch)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = c.SetClientProxyAddress(exch.ProxyAddress)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = c.Websocket.Setup(c.WsConnect,
+			c.Subscribe,
+			c.Unsubscribe,
+			exch.Name,
+			exch.Websocket,
+			exch.Verbose,
+			coinbeneWsURL,
+			exch.WebsocketURL,
+			exch.AuthenticatedWebsocketAPISupport)
+		if err != nil {
+			log.Fatal(err)
+		}
+		c.WebsocketConn = &wshandler.WebsocketConnection{
+			ExchangeName:         c.Name,
+			URL:                  c.Websocket.GetWebsocketURL(),
+			ProxyURL:             c.Websocket.GetProxyAddress(),
+			Verbose:              c.Verbose,
+			ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+			ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		}
+		c.Websocket.Orderbook.Setup(
+			exch.WebsocketOrderbookBufferLimit,
+			true,
+			true,
+			false,
+			false,
+			exch.Name)
+	}
+}
+
+// FetchTicker gets and stores ticker data for a currency pair
+func (c *Coinbene) FetchTicker(symbol string) (TickerResponse, error) {
+	var t TickerResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := fmt.Sprintf("%s%s%s?%s", c.APIUrl, coinbeneAPIVersion, coinbeneFetchTicker, params.Encode())
```

Comment:
To save you having to add ? on all these paths, you can use this https://github.com/thrasher-corp/gocryptotrader/blob/master/common/common.go#L434